### PR TITLE
feat(models): add Opus 4.7 + effort-tier model selector

### PR DESCRIPTION
**Pull request by dimakis**
## Summary
- Adds `claude-opus-4-7` and `claude-opus-4-7:max` (128k thinking budget) to the model selector
- Retains `claude-opus-4-6` as a previous-gen option
- New `parseModelSpec()` strips the `:max` effort suffix before sending the model ID to the SDK
- `resolveThinking()` maps `:max` → explicit 128k budget, default Opus stays adaptive
- Updates both mobile and desktop dropdowns

## Test plan
- [ ] Select Opus 4.7 from dropdown — verify adaptive thinking works
- [ ] Select Opus 4.7 Max — verify extended thinking with 128k budget
- [ ] Select Opus 4.6, Sonnet 4.6, Haiku 4.5 — verify no regression
- [ ] Verify `/api/models` returns all 5 entries
- [ ] Check Vertex availability for `claude-opus-4-7` (may not be GA yet)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/chat.ts [hunk 1]
```diff
@@ -93,11 +93,21 @@ export function getRepoConfig() {
...
+  { id: 'claude-opus-4-7', label: 'Opus 4.7', desc: 'Adaptive thinking' },
+  { id: 'claude-opus-4-7:max', label: 'Opus 4.7 Max', desc: 'Max thinking (128k)' },
+  { id: 'claude-opus-4-6', label: 'Opus 4.6', desc: 'Previous Opus' },
   { id: 'claude-sonnet-4-6', label: 'Sonnet 4.6', desc: 'Balanced' },
-  { id: 'claude-opus-4-6', label: 'Opus 4.6', desc: 'Most capable' },
   { id: 'claude-haiku-4-5', label: 'Haiku 4.5', desc: 'Fastest' },
 ];
 
+/** Split "claude-opus-4-7:max" → { model: "claude-opus-4-7", effort: "max" } */
+export function parseModelSpec(spec?: string): { model: string; effort: string | undefined } {
```
Comment (dimakis) [inline server/chat.ts:104]: 🟡 **missing_tests**: parseModelSpec is exported and contains non-trivial parsing logic (colon splitting, empty-string fallback) but has zero test coverage. Edge cases like trailing colon ('claude-opus-4-7:'), multiple colons, or empty effort string are untested. `[fixable]`

File: server/chat.ts [hunk 2]
```diff
@@ -122,8 +132,10 @@ function sdkEnv(): Record<string, string> {
 }
 
 function resolveThinking(
-  model?: string,
+  spec?: string,
 ): { type: 'adaptive' } | { type: 'enabled'; budgetTokens: number } | undefined {
+  const { model, effort } = parseModelSpec(spec);
+  if (model.includes('opus') && effort === 'max') return { type: 'enabled', budgetTokens: 128_000 };
```
Comment (dimakis) [inline server/chat.ts:138]: 🟡 **missing_tests**: resolveThinking was refactored to accept spec strings and gained a new code path (effort === 'max' → 128k budget). No tests cover the updated behavior or the interaction between parseModelSpec and resolveThinking. `[fixable]`

Comment (dimakis) [inline server/chat.ts:138]: 🔵 **unsafe_assumptions**: resolveThinking uses model.includes('opus') which will match any future model containing 'opus' in the name. Currently fine, but the effort tier ('max') is only meaningful for 4.7 — if a future opus model doesn't support the same budget, this would silently apply 128k thinking. Consider matching more precisely if this is a concern. `[fixable]`

File: frontend/src/pages/ChatView.tsx [hunk 3]
```diff
@@ -172,9 +172,11 @@ export function ChatView() {
           onChange={(e) => setModel(e.target.value)}
           disabled={messages.running}
         >
-          <option value="claude-sonnet-4-6">Sonnet</option>
-          <option value="claude-opus-4-6">Opus</option>
-          <option value="claude-haiku-4-5">Haiku</option>
+          <option value="claude-opus-4-7">Opus 4.7</option>
```
Comment (dimakis) [inline frontend/src/pages/ChatView.tsx:175]: 🟡 **regressions**: The default model in localStorage is 'claude-sonnet-4-6' (from constants.ts), but the first <option> is now 'claude-opus-4-7'. Existing users will still get Sonnet (correct), but new users or cleared-storage users will see Sonnet selected while the dropdown visually leads with Opus 4.7. This is cosmetic but may confuse users if intent was to default to the new model. `[fixable]`

Comment (dimakis) [inline frontend/src/pages/ChatView.tsx:175]: 🔵 **style**: Model options are duplicated verbatim between ChatView.tsx and DesktopChatView.tsx. The server already exposes GET /api/models (from AVAILABLE_MODELS). Consider driving the dropdown from that endpoint or a shared constant to avoid three-way sync. `[fixable]`

**Review by dimakis — COMMENTED** [review]
## Centaur Review

Found **5** issue(s) (3 warning).


### `server/chat.ts`
_Functional change looks correct but lacks unit tests for the new parseModelSpec/resolveThinking logic, and model options are hardcoded in three places._

- 🟡 **missing_tests** (L104): parseModelSpec is exported and contains non-trivial parsing logic (colon splitting, empty-string fallback) but has zero test coverage. Edge cases like trailing colon ('claude-opus-4-7:'), multiple colons, or empty effort string are untested. `[fixable]`
- 🟡 **missing_tests** (L138): resolveThinking was refactored to accept spec strings and gained a new code path (effort === 'max' → 128k budget). No tests cover the updated behavior or the interaction between parseModelSpec and resolveThinking. `[fixable]`
- 🔵 **unsafe_assumptions** (L138): resolveThinking uses model.includes('opus') which will match any future model containing 'opus' in the name. Currently fine, but the effort tier ('max') is only meaningful for 4.7 — if a future opus model doesn't support the same budget, this would silently apply 128k thinking. Consider matching more precisely if this is a concern. `[fixable]`


### `frontend/src/pages/ChatView.tsx`
_Functional change looks correct but lacks unit tests for the new parseModelSpec/resolveThinking logic, and model options are hardcoded in three places._

- 🟡 **regressions** (L175): The default model in localStorage is 'claude-sonnet-4-6' (from constants.ts), but the first <option> is now 'claude-opus-4-7'. Existing users will still get Sonnet (correct), but new users or cleared-storage users will see Sonnet selected while the dropdown visually leads with Opus 4.7. This is cosmetic but may confuse users if intent was to default to the new model. `[fixable]`
- 🔵 **style** (L175): Model options are duplicated verbatim between ChatView.tsx and DesktopChatView.tsx. The server already exposes GET /api/models (from AVAILABLE_MODELS). Consider driving the dropdown from that endpoint or a shared constant to avoid three-way sync. `[fixable]`

File: server/chat.ts [hunk 4]
```diff
@@ -122,8 +132,10 @@ function sdkEnv(): Record<string, string> {
...
   if (!model || model.includes('opus')) return { type: 'adaptive' };
```
Comment (dimakis) [inline server/chat.ts:139]: 🟡 **bugs**: resolveThinking: when spec is undefined, parseModelSpec returns { model: '', effort: undefined }. The empty string does not include 'opus', so the `!model` check (`!''` = true) short-circuits to adaptive thinking. This works but is fragile — the intent is 'no model specified → adaptive', but it relies on falsy empty string. A future refactor could break this. Consider an explicit early return for `!spec`. `[fixable]`

File: server/chat.ts (hunk 2 above)
Comment (dimakis) [inline server/chat.ts:138]: 🔵 **unsafe_assumptions**: The effort tier is matched with strict equality (`effort === 'max'`), but there is no validation that effort is a known value. If a client sends 'claude-opus-4-7:turbo' or any arbitrary suffix, it silently falls through to adaptive thinking. This is acceptable for now but worth documenting or validating as more tiers are added. `[fixable]`

File: frontend/src/lib/constants.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ export const WS_RECONNECT_POLL_MS = 5_000;
 export const SCROLL_NEAR_BOTTOM_PX = 150;
 export const SCROLL_RESTORE_DELAY_MS = 100;
 export const LAST_SESSION_KEY = 'mitzo-last-session';
-export const DEFAULT_MODEL = 'claude-sonnet-4-6';
+export const DEFAULT_MODEL = 'claude-opus-4-7';
```
Comment (dimakis) [inline frontend/src/lib/constants.ts:10]: 🔵 **regressions**: DEFAULT_MODEL changed from 'claude-sonnet-4-6' to 'claude-opus-4-7'. Existing users with no localStorage preference will silently switch to a more expensive model. This is likely intentional but worth noting in the PR description.

**Review by dimakis — COMMENTED** [review]
## Centaur Review

Found **5** issue(s) (2 warning).


### `server/chat.ts`
_Functional changes are correct but the new parseModelSpec/resolveThinking logic lacks unit tests (TDD violation), and the model list is duplicated across three files._

- 🟡 **missing_tests**: parseModelSpec() and the updated resolveThinking() have zero test coverage. parseModelSpec is a pure function with clear edge cases (no colon, colon at end, multiple colons, undefined input) that should be trivially testable. resolveThinking now has branching logic for effort tiers that is also untestable without unit tests. This violates the project's TDD requirement. `[fixable]`
- 🟡 **bugs** (L139): resolveThinking: when spec is undefined, parseModelSpec returns { model: '', effort: undefined }. The empty string does not include 'opus', so the `!model` check (`!''` = true) short-circuits to adaptive thinking. This works but is fragile — the intent is 'no model specified → adaptive', but it relies on falsy empty string. A future refactor could break this. Consider an explicit early return for `!spec`. `[fixable]`
- 🔵 **unsafe_assumptions** (L138): The effort tier is matched with strict equality (`effort === 'max'`), but there is no validation that effort is a known value. If a client sends 'claude-opus-4-7:turbo' or any arbitrary suffix, it silently falls through to adaptive thinking. This is acceptable for now but worth documenting or validating as more tiers are added. `[fixable]`


### `frontend/src/pages/ChatView.tsx`
_Functional changes are correct but the new parseModelSpec/resolveThinking logic lacks unit tests (TDD violation), and the model list is duplicated across three files._

- 🔵 **style**: Model options are hardcoded in both ChatView.tsx and DesktopChatView.tsx, duplicating the AVAILABLE_MODELS list from server/chat.ts. The server already exposes GET /api/models — consider fetching from it or extracting a shared constant to avoid three-way sync drift. `[fixable]`


### `frontend/src/lib/constants.ts`
_Functional changes are correct but the new parseModelSpec/resolveThinking logic lacks unit tests (TDD violation), and the model list is duplicated across three files._

- 🔵 **regressions** (L10): DEFAULT_MODEL changed from 'claude-sonnet-4-6' to 'claude-opus-4-7'. Existing users with no localStorage preference will silently switch to a more expensive model. This is likely intentional but worth noting in the PR description.
